navigation/localization/auv_localization/auv_localization/auv_localization.py

```diff
@@ -0,0 +1,220 @@
+        observation[2] = self.depth
+
+        if not self.first_measurement:
+            self.X_auv_relative = self.get_first_measured_state(observation)
+            self.kf_auv.initialize_state(self.X_auv_relative, self.Q_auv_relative, self.R_auv_relative, self.P_auv_relative)
+            self.first_measurement = True
+            # self.get_logger().info(f"initializaing: {self.X_auv_relative }") debugging....
+        else:
+            self.X_auv_relative, self.P_auv_relative = self.kf_auv.estimate(observation, feedback_image=True)
+            self.get_logger().info(f"estimating relative position: {self.X_auv_relative }")
+        if self.X_auv_relative is not None:
+            auv_position = self.get_world_position(self.X_auv_relative[:3])
+            self.publish_auv_position(auv_position)
+            self.publish_tf(auv_position)
+
+    def get_first_measured_state(self, observation):
+        direction_vector = np.dot(self.K_inv, observation)
+        direction_world = np.dot(np.eye(3), direction_vector)  # Using an identity matrix for R_wa
+        scale = self.depth / direction_world[2]
+        vector_auv_drone = direction_world * scale
+        return np.array([vector_auv_drone, np.zeros(3)]).flatten().T
+
+    def publish_tf(self, position, orientation=None):
+        # Create a TransformStamped message
+        t = TransformStamped()
+
+        # Set the frame information
+        t.header.stamp = self.get_clock().now().to_msg()
+        t.header.frame_id = self.map_frame  # Parent frame
+        t.child_frame_id = 'auv_estimated_state'  # Child frame for the estimated state
+
+        # Set the translation (position) in the TF message
+        t.transform.translation.x = position[0]
+        t.transform.translation.y = position[1]
+        t.transform.translation.z = position[2]
+
+        # Check if orientation is provided; if not, set it to identity quaternion (no rotation)
+        if orientation:
+            t.transform.rotation.x = orientation[0]
+            t.transform.rotation.y = orientation[1]
+            t.transform.rotation.z = orientation[2]
+            t.transform.rotation.w = orientation[3]
+        else:
+            # Default to no rotation if no orientation is provided
+            t.transform.rotation.x = 0.0
+            t.transform.rotation.y = 0.0
+            t.transform.rotation.z = 0.0
+            t.transform.rotation.w = 1.0  # Identity quaternion (no rotation)
+
+        # Broadcast the transform
+        self.tf_broadcaster.sendTransform(t)
+
+    def publish_auv_position(self, position):
+        # self.get_logger().info(f"position being published : {position}")
+        
+        # Create an Odometry message
+        odom_msg = Odometry()
+        
+        # Set header information
+        odom_msg.header.stamp = self.get_clock().now().to_msg()  # Add timestamp
+        odom_msg.header.frame_id = "odom"  # Typically, "odom" frame for localization
```

Comment:
just "odom" is useless. must be namespaced with robot name. the estimate of the auv position is wrt to drone's odom frame, right?

Reply:
it is relative position : (auv_position)_{world_frame} so I should publish it as '{robot_name}/detection/estimated_odom'?

Reply:
One, it shouldnt be in "map_gt", it should be in the same map frame the drone happens to be. 
And yes, `{robot_name}/auv/odom` seems good 👌